sql/distinct.go

```diff
@@ -37,6 +37,9 @@ func (*planner) distinct(n *parser.Select, p planNode) planNode {
 		d.columnsInOrder = make([]bool, len(p.Columns()))
 	}
 	for _, p := range p.Ordering() {
+		if p == 0 {
```

Comment:
Yeah, perhaps we would gain some clarity but using the type system more aggressively rather than encoding sophisticated meaning into primitive values. It's still not clear to me why `break` is used below rather than `continue`.